src/storage/storage.cpp

```diff
@@ -840,6 +956,86 @@ void Storage::PopulateData(const DataLayout &layout, char *memory_ptr)
+                layout.GetBlockPtr<partition::MultiLevelPartition::LevelData, true>(
+                    memory_ptr, DataLayout::MLD_LEVEL_DATA);
+            auto mld_partition_ptr = layout.GetBlockPtr<partition::PartitionID, true>(
+                memory_ptr, DataLayout::MLD_PARTITION);
+            auto mld_chilren_ptr = layout.GetBlockPtr<partition::CellID, true>(
+                memory_ptr, DataLayout::MLD_CELL_TO_CHILDREN);
+
+            io::FileReader reader(config.mld_partition_path, io::FileReader::VerifyFingerprint);
+
+            reader.ReadInto(mld_level_data_ptr);
+            reader.ReadInto(mld_partition_ptr, layout.GetBlockEntries(DataLayout::MLD_PARTITION));
+            reader.ReadInto(mld_chilren_ptr,
+                            layout.GetBlockEntries(DataLayout::MLD_CELL_TO_CHILDREN));
+        }
+
+        if (boost::filesystem::exists(config.mld_storage_path))
+        {
+            io::FileReader reader(config.mld_storage_path, io::FileReader::VerifyFingerprint);
+            auto mld_cell_weights_ptr =
+                layout.GetBlockPtr<EdgeWeight, true>(memory_ptr, DataLayout::MLD_CELL_WEIGHTS);
+            auto mld_source_boundary_ptr =
+                layout.GetBlockPtr<NodeID, true>(memory_ptr, DataLayout::MLD_CELL_SOURCE_BOUNDARY);
+            auto mld_destination_boundary_ptr = layout.GetBlockPtr<NodeID, true>(
+                memory_ptr, DataLayout::MLD_CELL_DESTINATION_BOUNDARY);
+            auto mld_cells_ptr = layout.GetBlockPtr<partition::CellStorage::CellData, true>(
+                memory_ptr, DataLayout::MLD_CELLS);
+            auto mld_cell_level_offsets_ptr = layout.GetBlockPtr<std::uint64_t, true>(
+                memory_ptr, DataLayout::MLD_CELL_LEVEL_OFFSETS);
+
+            reader.ReadInto(mld_cell_weights_ptr,
+                            layout.GetBlockEntries(DataLayout::MLD_CELL_WEIGHTS));
+            reader.ReadInto(mld_source_boundary_ptr,
+                            layout.GetBlockEntries(DataLayout::MLD_CELL_SOURCE_BOUNDARY));
+            reader.ReadInto(mld_destination_boundary_ptr,
+                            layout.GetBlockEntries(DataLayout::MLD_CELL_DESTINATION_BOUNDARY));
+            reader.ReadInto(mld_cells_ptr, layout.GetBlockEntries(DataLayout::MLD_CELLS));
+            reader.ReadInto(mld_cell_level_offsets_ptr,
+                            layout.GetBlockEntries(DataLayout::MLD_CELL_LEVEL_OFFSETS));
+        }
+
+        if (boost::filesystem::exists(config.edge_based_graph_path))
+        {
+            io::FileReader reader(config.edge_based_graph_path, io::FileReader::VerifyFingerprint);
+
+            const auto number_of_edges = reader.ReadElementCount64();
+            const auto number_of_nodes = reader.ReadOne<EdgeID>() + 1;
+            std::vector<extractor::EdgeBasedEdge> original_edges(number_of_edges);
+            reader.ReadInto(original_edges);
+
+            original_edges = partition::splitBidirectionalEdges(std::move(original_edges));
```

Comment:
Missed this part. For the future we would want to not do this on load-time but before the graph hit's disk in `osrm-extract`. This would avoid over-allocating. But since this will need changes to how `osrm-contract` works (currently iterating over the edge list), it is fine to add a `TODO` here. Can you open a ticket to capture this?